processor/src/main/java/com/linecorp/decaton/processor/runtime/ProcessorSubscription.java

```diff
@@ -103,14 +101,22 @@ private void updateState(SubscriptionStateListener.State newState) {
                      .collect(Collectors.toSet());
     }
 
-    private void commitCompletedOffsets(Consumer<?, ?> consumer) {
+    private void commitCompletedOffsets(Consumer<?, ?> consumer, boolean sync) {
         Map<TopicPartition, OffsetAndMetadata> commitOffsets = contexts.commitOffsets();
         if (commitOffsets.isEmpty()) {
-            logger.debug("No new offsets to commit, skipping commit");
+            log.debug("No new offsets to commit, skipping commit");
             return;
         }
-        logger.debug("Committing offsets: {}", commitOffsets);
-        consumer.commitSync(commitOffsets);
+        log.debug("Committing offsets {}: {}", sync ? "SYNC" : "ASYNC", commitOffsets);
+        if (sync) {
+            consumer.commitSync(commitOffsets);
+        } else {
+            consumer.commitAsync(commitOffsets, (offsets, exception) -> {
+                if (exception != null) {
+                    log.warn("Offset commit failed asynchronously", exception);
+                }
+            });
+        }
         contexts.updateCommittedOffsets(commitOffsets);
```

Comment:
I think I vaguely understand, but can you comment as to why this is synchronous and not a callback of `commitAsync` ?

Reply:
Oh, I think this should be handled differently for async case (do in callback). Let me confirm and fix, thanks for nice catch!